src/cryptsetup/cryptsetup-generator.c

```diff
@@ -549,7 +586,68 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 free_and_replace(d->keyfile, keyfile);
                 free_and_replace(d->keydev, keydev);
+        } else if (streq(key, "luks.hdrdev")) {
+                size_t n;
+                _cleanup_free_ char *hdrdev = NULL;
+
+                if (proc_cmdline_value_missing(key, value))
+                        return 0;
+
+                n = strspn(value, ALPHANUMERICAL "-");
+                if (value[n] != '=') {
+                        log_warning("Failed to parse luks.hdrdev= kernel command line switch. UUID is invalid, ignoring.");
+                        return 0;
+                }
+
+                uuid = strndup(value, n);
+                if (!uuid)
+                        return log_oom();
+
+                if (!id128_is_valid(uuid)) {
+                        log_warning("Failed to parse luks.hdrdev= kernel command line switch. UUID is invalid, ignoring.");
+                        return 0;
+                }
+
+                d = get_crypto_device(uuid);
+                if (!d)
+                        return log_oom();
+
+                hdrdev = fstab_node_to_udev_node(value + n + 1);
+                if (!hdrdev)
+                        return log_oom();
+
+                free_and_replace(d->hdrdev, hdrdev);
+        } else if (streq(key, "luks.data")) {
```

Comment:
I am not convinced about the naming. Both "luks.data" and "luks.hdrdev" take a path to a device in some form, but one contains "dev" and one doesn't in the name?

Also, I am not fond of needlessly abbreviations such as "hdr", in particular in options that are probably only going to be used in a few corner cases, and not in the common case. Please don't needlessly abbreivate.

Maybe use luks.data_dev= and luks.header_dev=?

Reply:
Ok, should be fixed